106-auto-rebalance-on-imbalanced-clusters.md

```diff
@@ -0,0 +1,471 @@
+# Auto-rebalance on imbalanced clusters
+
+This proposal is about adding support for auto-rebalancing the Kafka cluster in case it gets imbalanced due to some issues like unevenly distributed replicas or overloaded brokers e.t.c.
+When enabled, the Strimzi operator should automatically resolve these issues detected by the Anomaly Detector Manager by running KafkaRebalance via Cruise Control using the KafkaRebalance resource.
+Anomalies are detected by Cruise Control using the anomaly detector manager (see section [ Anomaly Detector Manager](./106-auto-rebalance-on-imbalanced-clusters.md#anomaly-detector-manager) below for a detailed description).
+
+## Motivation
+
+Currently, any anomaly that the user is notified about would need to be fixed manually by using the `KafkaRebalance` custom resource.
+With smaller clusters, it is feasible to fix things manually. However, for larger ones it can be very time-consuming, or just not feasible, to fix all the anomalies on your own.
+It would be useful for users of Strimzi to be able to have these anomalies fixed automatically whenever they are detected.
+
+### Introduction to Self Healing
+
+![self-healing flow diagram](images/106-self-healing-flow.png)
+
+The above flow diagram depicts the self-healing process in Cruise Control.
+The anomaly detector manager detects an anomaly (using the detector classes) and forwards it to the notifier.
+The configured notifiers provides alerts to the users about the detected anomaly and also returns the action that needs to be taken on the anomaly i.e. whether to fix it, ignore it or delay it.
```

Comment:
The alert mechanism isn't out of the box. A notifier can have its own logic without generating any alerts. Even just triggering the fix without notifying anyone what's happening. So there is no assumption that a "configured" notifier provides alerts. I think this sentence should say that the notifier makes the decision about the action to take. Then CC provides some notifiers which are able to alert the user in several ways (MS Teams, Slack, etc etc).